CONTRIBUTING.md

```diff
@@ -221,6 +221,8 @@ E2E are most useful for testing what happens on a page after it has been loaded
 
 Translations help make Starlight accessible to more people.
 
+Check out the dedicated [i18n contribution guide](https://contribute.docs.astro.build/guides/i18n/#quality-standards--adaptation) for more details regarding our translation process and quality standards.
```

Comment:
Perhaps note that this is in the Astro guide like you did for the other link in this PR to help give some context to the reader?